src/main/java/edu/ksu/canvas/errors/GenericErrorHandler.java

```diff
@@ -0,0 +1,42 @@
+package edu.ksu.canvas.errors;
+
+import com.google.gson.Gson;
+import edu.ksu.canvas.exception.CanvasException;
+import edu.ksu.canvas.impl.GsonResponseParser;
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpResponse;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+/**
+ * The error handler that should be used when creation of a Canvas object fails.
+ */
+public class GenericErrorHandler implements ErrorHandler {
+
+    private Pattern coursePattern = Pattern.compile("/api/v1/accounts/\\d+/courses");
+    private Pattern loginPattern = Pattern.compile("/api/v1/accounts/\\d+/logins");
```

Comment:
The URLs can have SIS IDs in them like `sis_course_id:an-example-sis-id` which wouldn't match on just the digits.

Reply:
https://canvas.instructure.com/doc/api/file.object_ids.html

Reply:
Ok, I thought we weren't really covering all the cases on this PR, just adding a couple more that I had already tested. You proposed to move from this pattern check to every 400 error in the future, should I take a look to it now?

Reply:
I'll change it for "/accounts/xx/courses" but "/courses/yy/whatever" won't be covered as it is, until we add more patterns or catch the 400s.